civis/futures.py

```diff
@@ -299,6 +298,10 @@ def cancel(self):
             return False
 
 
+def _format_job_run_ids_in_exception(job_id: int, run_id: int) -> str:
+    return f"(Job ID {job_id} / Run ID {run_id})"
```

Comment:
based on the ticket description, I'm not sure if the job id and run id were meant to be added to the error description or as an attribute to the `CivisJobFailure` class. On one had I think it would be easy to reference if stored in the error class. ie.
```
try:
...
except CivisJobFailure as exc:
  log.error(f'Job {exc.job_id} failed please re-run')
```
But that information should be stored and accessible on the future already, so maybe I don't understand the use case.

I still think this formatting might be better to have on the error class. Maybe format the `__str__` to prefix the error with `Job #{job_id} / Run #{run_id} failed. {self.error_message}`. What do you think?

Reply:
it would also be nice to account for the case when run_id is None.

Reply:
Good question re: job/run id as attributes of `CivisJobFailure` versus just part of the error message string. While power users would know the future already has the job/run id (like you've mentioned) and would therefore know what to do, less advanced users might want/hope to see an unhandled CivisJobFailure come with the job/run id in the logging to aid debugging. Which is why I've been leaning towards fiddling with the error msg string with the job/run id included in some way.

I like your suggestions. I'm going to make the following updates:
* Add job/run id as attributes to `CivisJobFailure`. No harm having these even though the future already does. Also, it'd be handy to have these attributes for the next bullet point...
* Move the job/run id string formatting code to under `CivisJobfailure`. I'm considering updating `self.error_message` directly so that it always has the job/run id embedded, which would mean `__str__` is also taken care of.
* If run_id is None, don't show the unwieldy `/ run ID None` bit in the error msg.

Reply:
thanks for the explanation, whatever you think is best! I'm also okay with the original way if it avoids too much complication with the error messages